tdrs-backend/tdpservice/users/api/login.py

```diff
@@ -207,3 +251,102 @@ def get(self, request, *args, **kwargs):
+
+        except ValueError as e:
+            logger.exception(e)
+            return Response(
+                {
+                    "error": str(e)
+                },
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+    def get_auth_options(self, access_token, sub):
+        """Add specific auth properties for the CustomAuthentication handler."""
+        auth_options = {"login_gov_uuid": sub}
+        return auth_options
+
+
+class TokenAuthorizationAMS(TokenAuthorizationOIDC):
+    """Define methods for handling login request from HHS AMS."""
+
+    def decode_payload(self, token_data, options=None):
+        """Decode the payload with keys for AMS."""
+        id_token = token_data.get("id_token")
+        access_token = token_data.get("access_token")
+
+        ams_configuration = LoginRedirectAMS.get_ams_configuration()
+        certs_endpoint = ams_configuration["jwks_uri"]
+        cert_str = generate_jwt_from_jwks(certs_endpoint)
+        issuer = ams_configuration["issuer"]
+        audience = settings.AMS_CLIENT_ID
+
+        decoded_id_token = self.decode_jwt(id_token, issuer, audience, cert_str, {"verify_aud": False})
+        decoded_access_token = self.decode_jwt(access_token, issuer, audience, cert_str, {"verify_aud": False})
+
+        return {
+            "id_token": decoded_id_token,
+            "access_token": access_token,
+            "decoded_access_token": decoded_access_token
+        }
+
+    def get_token_endpoint_response(self, code):
+        """Build out the query string params and full URL path for token endpoint."""
+        # First fetch the token endpoint from AMS.
+        ams_configuration = LoginRedirectAMS.get_ams_configuration()
+        options = {
+            "client_id": settings.AMS_CLIENT_ID,
+            "client_secret": settings.AMS_CLIENT_SECRET,
+            "scope": "openid+email+profile",
+            "code": code,
+            "grant_type": "authorization_code",
+            "redirect_uri": settings.BASE_URL + "/oidc/ams",
+        }
+
+        token_endpoint = ams_configuration["token_endpoint"]
+
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+
+        return requests.post(token_endpoint, headers=headers, data=options)
+
+    def get_auth_options(self, access_token, sub):
+        """Add specific auth properties for the CustomAuthentication handler."""
+        logger.info(access_token)
+        if access_token:
+            auth_options = {}
+            # Fetch userinfo endpoint for AMS to authenticate against hhsid, or
+            # other user claims.
+            ams_configuration = LoginRedirectAMS.get_ams_configuration()
+            userinfo_response = requests.post(ams_configuration["userinfo_endpoint"],
+                                              {"access_token": access_token})
+            user_info = userinfo_response.json()
+            logger.info(user_info)
+            # TODO Use `hhs_id` as primary authentication key
```

Comment:
Could you expand this TODO description to let people know we are blocked by AMS currently? You could add a link to your comment [here](https://github.com/raft-tech/TANF-app/issues/1136#issuecomment-996822564).